etcd/v1alpha1/types_pacemakercluster.go

```diff
@@ -0,0 +1,489 @@
+	// Kept for debugging purposes only; healthcheck should not need to parse this.
+	// When present, it must be between 1 and 262144 characters long (max 256KB). This limit protects the API from XML bombs and excessive memory consumption.
+	// When not present, the raw XML output is not available.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=262144
+	// +optional
+	RawXML string `json:"rawXML,omitempty"`
+
+	// collectionError contains any error encountered while collecting status
+	// When present, it must be between 1 and 2048 characters long (max 2KB). This limit ensures that the error message can be displayed in a human-readable format.
+	// When not present, no collection errors are available and the status collection is assumed to be successful.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	// +optional
+	CollectionError string `json:"collectionError,omitempty"`
+
+	// summary provides high-level counts and flags for the cluster state
+	// When present, it must be a valid PacemakerSummary object.
+	// When not present, the summary is not available. This likely indicates that there is an error parsing the raw XML output.
+	// +optional
+	Summary *PacemakerSummary `json:"summary,omitempty"`
+
+	// nodes provides detailed information about each node in the cluster
+	// When present, it must be a list of 1 or 2 PacemakerNodeStatus objects. Two is expected in a healthy cluster.
+	// When not present, the nodes are not available. This likely indicates that there is an error parsing the raw XML output.
+	// If only one node is present, this indicates that the cluster is in the process of replacing a failed node.
+	// +listType=atomic
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=2
+	// +optional
+	Nodes []PacemakerNodeStatus `json:"nodes,omitempty"`
+
+	// resources provides detailed information about each resource in the cluster
+	// When present, it must be a list of 1 or more PacemakerResourceStatus objects.
+	// When not present, the resources are not available. This likely indicates that there is an error parsing the raw XML output.
+	// The number of resources is expected to be between 1 and 16, but is most likely to be exactly 6.
+	// The critical resources that expect to run on both nodes are: kubelet, etcd, and a fencing resource (i.e. redfish) for each node.
+	// This could drift over time as Two Node Fencing matures, so this is left flexible.
+	// +listType=atomic
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=16
+	// +optional
+	Resources []PacemakerResourceStatus `json:"resources,omitempty"`
+
+	// nodeHistory provides recent operation history for troubleshooting
+	// When present, it must be a list of 1 or more PacemakerNodeHistoryEntry objects.
+	// When not present, the node history is not available. This is the expected status for a healthy cluster.
```

Comment:
I wouldn't think that an empty node history is expected status.  Assuming that this is basically `<node_history/>` from `crm_mon`, you should have a tree for all the start and monitor operations for every resource on each node.  If there's no history, I would assume there's no running resources.

Reply:
The reason we allow this to be empty is that we only push up "recent" information. Basically, we are trying to collect the information from pacemaker that would indicate that we've gone off the rails. So before this API is invoked, we gather all of the information, then filter out any node history event that isn't within the last 5 minutes.

For fencing history, we carry failures longer - an hour long context window.

So it's not a full 1-1 mapping. :) I'll make a note of that in the API.